apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.ts

```diff
@@ -18,37 +19,79 @@ router.get(
     const courseInfoExists = await fs.pathExists(
       path.join(res.locals.course.path, 'infoCourse.json'),
     );
+    const availableTimezones = await getAvailableTimezones();
 
     res.send(
-      InstructorCourseAdminSettings({ resLocals: res.locals, coursePathExists, courseInfoExists }),
+      InstructorCourseAdminSettings({
+        resLocals: res.locals,
+        coursePathExists,
+        courseInfoExists,
+        availableTimezones,
+      }),
     );
   }),
 );
 
-router.post('/', (req, res, next) => {
-  if (!res.locals.authz_data.has_course_permission_edit || res.locals.course.example_course) {
-    return next(
-      error.make(403, 'Access denied (must be course editor and must not be example course)'),
-    );
-  }
-
-  if (req.body.__action === 'add_configuration') {
-    const editor = new CourseInfoEditor({
-      locals: res.locals,
-    });
-    editor.canEdit((err) => {
-      if (ERR(err, next)) return;
-      editor.doEdit((err, job_sequence_id) => {
-        if (ERR(err, (e) => logger.error('Error in doEdit()', e))) {
-          res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
-        } else {
-          res.redirect(req.originalUrl);
-        }
+router.post(
+  '/',
+  asyncHandler(async (req, res, next) => {
+    if (!res.locals.authz_data.has_course_permission_edit || res.locals.course.example_course) {
+      return next(
+        error.make(403, 'Access denied (must be course editor and must not be example course)'),
+      );
+    }
+
+    if (req.body.__action === 'update_configuration') {
+      const courseInfo = JSON.parse(
+        await fs.readFileSync(path.join(res.locals.course.path, 'infoCourse.json'), 'utf8'),
```

Comment:
Commenting before I forget: one should basically never use `*Sync` variants of `fs` functions in production code, since they'll block the event loop.

Reply:
Updated  all uses of `readFileSync` to `readFile` in [79f4f05](https://github.com/PrairieLearn/PrairieLearn/pull/9581/commits/79f4f05b8d78db39a94b29e232a22bdc4c552b34). Also, thanks for the reading on these and learning about the event loop in Node!